klippy/extras/tmc5160.py

```diff
@@ -0,0 +1,437 @@
+}
+fields["FACTORY_CONF"] = {
+    "FACTORY_CONF":             0x1F << 0
+}
+fields["GCONF"] = {
+    "recalibrate":              0x01 << 0,
+    "faststandstill":           0x01 << 1,
+    "en_pwm_mode":              0x01 << 2,
+    "multistep_filt":           0x01 << 3,
+    "shaft":                    0x01 << 4,
+    "diag0_error":              0x01 << 5,
+    "diag0_otpw":               0x01 << 6,
+    "diag0_stall":              0x01 << 7,
+    "diag1_stall":              0x01 << 8,
+    "diag1_index":              0x01 << 9,
+    "diag1_onstate":            0x01 << 10,
+    "diag1_steps_skipped":      0x01 << 11,
+    "diag0_int_pushpull":       0x01 << 12,
+    "diag1_poscomp_pushpull":   0x01 << 13,
+    "small_hysteresis":         0x01 << 14,
+    "stop_enable":              0x01 << 15,
+    "direct_mode":              0x01 << 16,
+    "test_mode":                0x01 << 17
+}
+fields["GSTAT"] = {
+    "reset":                    0x01 << 0,
+    "drv_err":                  0x01 << 1,
+    "uv_cp":                    0x01 << 2
+}
+fields["IHOLD_IRUN"] = {
+    "IHOLD":                    0x1F << 0,
+    "IRUN":                     0x1F << 8,
+    "IHOLDDELAY":               0x0F << 16
+}
+fields["IOIN"] = {
+    "REFL_STEP":                0x01 << 0,
+    "REFR_DIR":                 0x01 << 1,
+    "ENCB_DCEN_CFG4":           0x01 << 2,
+    "ENCA_DCIN_CFG5":           0x01 << 3,
+    "DRV_ENN":                  0x01 << 4,
+    "ENC_N_DCO_CFG6":           0x01 << 5,
+    "SD_MODE":                  0x01 << 6,
+    "SWCOMP_IN":                0x01 << 7,
+    "VERSION":                  0xFF << 24
+}
+fields["LOST_STEPS"] = {
+    "LOST_STEPS":               0xfffff << 0
+}
+fields["MSCNT"] = {
+    "MSCNT":                    0x3ff << 0
+}
+fields["MSCURACT"] = {
+    "CUR_A":                    0x1ff << 0,
+    "CUR_B":                    0x1ff << 16
+}
+fields["OTP_READ"] = {
+    "OTP_FCLKTRIM":             0x1f << 0,
+    "otp_S2_LEVEL":             0x01 << 5,
+    "otp_BBM":                  0x01 << 6,
+    "otp_TBL":                  0x01 << 7
+}
+fields["PWM_AUTO"] = {
+    "PWM_OFS_AUTO":             0xff << 0,
+    "PWM_GRAD_AUTO":            0xff << 16
+}
+fields["PWMCONF"] = {
+    "PWM_OFS":                  0xFF << 0,
+    "PWM_GRAD":                 0xFF << 8,
+    "pwm_freq":                 0x03 << 16,
+    "pwm_autoscale":            0x01 << 18,
+    "pwm_autograd":             0x01 << 19,
+    "freewheel":                0x03 << 20,
+    "PWM_REG":                  0x0F << 24,
+    "PWM_LIM":                  0x0F << 28
+}
+fields["PWM_SCALE"] = {
+    "PWM_SCALE_SUM":            0xff << 0,
+    "PWM_SCALE_AUTO":           0x1ff << 16
+}
+fields["TPOWERDOWN"] = {
+    "TPOWERDOWN":               0xff << 0
+}
+fields["TPWMTHRS"] = {
+    "TPWMTHRS":                 0xfffff << 0
+}
+fields["TSTEP"] = {
+    "TSTEP":                    0xfffff << 0
+}
+
+FieldFormatters = tmc2130.FieldFormatters
```

Comment:
This needs to be wrapped in a dict() call to make a copy of tmc2130.FieldFormatters - otherwise, the update() call below will modify the contents of the tmc2130.py module.

Reply:
done